### PR TITLE
source-zoho-desk contribution from itsxdamdam

### LINKED ISSUE #46645: [connector-request] Zoho Desk
This is a **suggestion** to create a connector for **Zoho Desk** using the UI Connector Builder.
The website for the connector is https://www.zoho.com/desk.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by itsxdamdam**
## What
Closes #46645 

This PR adds a new connector for source Zoho Desk (source-zoho-desk).

Contributor-provided connector description:

This directory contains the manifest-only connector for source-zoho-desk
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |
| `token_refresh_endpoint` | `string` | Token Refresh Endpoint.  |  |
| `refresh_token` | `string` | OAuth Refresh Token.  |  |
| `include_custom_domain` | `boolean` | include Custom Domain.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| organization | id | No pagination | ✅ |  ❌  |
| all_organizations | id | No pagination | ✅ |  ❌  |
| accessible_organizations |  | No pagination | ✅ |  ❌  |
| agent |  | No pagination | ✅ |  ❌  |
| list_agents | id | DefaultPaginator | ✅ |  ❌  |
| agent_details_by_id | id | No pagination | ✅ |  ❌  |
| profiles | id | No pagination | ✅ |  ❌  |
| list_roles | id | DefaultPaginator | ✅ |  ❌  |
| teams | id | No pagination | ✅ |  ❌  |
| team_members | id | No pagination | ✅ |  ❌  |
| list_departments |  | DefaultPaginator | ✅ |  ❌  |
| channels | code | No pagination | ✅ |  ❌  |
| list_tickets | id | DefaultPaginator | ✅ |  ❌  |
| list_all_threads | id | DefaultPaginator | ✅ |  ❌  |
| get_latest_thread | id | No pagination | ✅ |  ❌  |
| list_contacts | id | DefaultPaginator | ✅ |  ❌  |
| webhooks |  | No pagination | ✅ |  ❌  |
| list_accounts | id | DefaultPaginator | ✅ |  ❌  |
| list_contracts | id | DefaultPaginator | ✅ |  ❌  |
| list_tasks | id | DefaultPaginator | ✅ |  ❌  |
| list_products | id | DefaultPaginator | ✅ |  ❌  |
| list_articles | id | No pagination | ✅ |  ❌  |
| list_events | id | DefaultPaginator | ✅ |  ❌  |
| modules | id | No pagination | ✅ |  ❌  |
| list_users | id | DefaultPaginator | ✅ |  ❌  |
| ticket_activities | id | DefaultPaginator | ✅ |  ❌  |
| list_attachments |  | DefaultPaginator | ✅ |  ❌  |
| product | id | No pagination | ✅ |  ❌  |
| task_attachments | id | DefaultPaginator | ✅ |  ❌  |
| list_calls | id | DefaultPaginator | ✅ |  ❌  |
| call | id | No pagination | ✅ |  ❌  |
| list_call_comments | id | DefaultPaginator | ✅ |  ❌  |
| dashboard_created_tickets |  | No pagination | ✅ |  ❌  |
| list_user_groups | id | DefaultPaginator | ✅ |  ❌  |
| dashboard_onhold_tickets | value | No pagination | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

**Comment by itsxdamdam** [comment]
Making a new PR for this shortly.

File: docs/integrations/sources/zoho-desk.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+| `agent_ids` | `string` | agent_ids. agent ids |  |
+| `profile_id` | `string` | profile_id. Profile ID |  |
+| `limit` | `integer` | limit. Number of roles to display. The default value is 15 and the maximum value allowed is 500. |  |
+| `isvisible` | `boolean` | isVisible.  Key that filters roles according to their visibility in the UI |  |
+| `isdefault` | `boolean` | isDefault. Key that denotes whether the roles must be default roles or custom roles |  |
+| `searchstr` | `string` | searchStr. String to search for roles by name or description. The string must contain at least one character. Three search methods are supported: 1) string* - Searches for roles whose name or description start with the string, 2) *string* - Searches for roles whose name or description contain the string, 3) string - Searches for roles whose name or description is an exact match for the string |  |
+| `team_id` | `string` | team_id. Team ID |  |
+| `department_id` | `string` | department_id. Department Id |  |
+| `ticket_id` | `string` | ticket_id. ticket id |  |
+| `account_id` | `string` | account_id. Account ID |  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/zoho-desk.md:25]: I don't believe the connector should have so many parameters. Usually API returns all values if you don't specify a value like status, isconfirmed etc.. 
Why are you asking for parameters you can use Parent streams as department, tickets and agents?

File: docs/integrations/sources/zoho-desk.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| organization |  | No pagination | ✅ |  ❌  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/zoho-desk.md:30]: Why organizations and organization? What is different from both streams? The same for agent and agents...

File: airbyte-integrations/connectors/source-zoho-desk/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,1388 @@
...
+          - "null"
+      zip:
+        type:
+          - string
+          - "null"
+  organizations:
+    type: object
+    $schema: http://json-schema.org/draft-07/schema#
+    additionalProperties: true
+    properties: {}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zoho-desk/manifest.yaml:1268]: Schema doesn't look right.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments.

File: docs/integrations/sources/zoho-desk.md (hunk 1 above)
Comment (itsxdamdam) [reply docs/integrations/sources/zoho-desk.md]: Yes. I made a lot of corrections on the new PR. I just found out about Parent streams

**Comment by marcosmarxm** [comment]
Please sign the CLA and make sure credentials are valid. Moving new review date to next week. Please let me know credentials are valid again.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Please use Oauth auth method instead of Bearer token expires every hour. If this is a hard limitation I don't think this connector can be accepted.

Second, all pagination is wrong:
image
The same applies to `limit`. 


Please all of these to all streams, run tests and make sure they're properly working.
I'm moving the next review to this contribution to next week.

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
@itsxdamdam need you sign the CLA to merge the contribution

**Comment by itsxdamdam** [comment]
> @itsxdamdam need you sign the CLA to merge the contribution

Hi Marcos, How do i do that?

**Comment by marcosmarxm** [comment]
Check first comment in the PR contribution it has a link to sign the CLA
https://github.com/airbytehq/airbyte/pull/46863#issuecomment-2408594933

**Comment by itsxdamdam** [comment]
signed!

**Pull request merged**